### ISSUE TITLE
[Bug] How to build

### ISSUE THREAD

**Issue by zlokomatic** [cloudberrydb/pljava#1]
### Cloudberry Database and PL/Java version

main branch from git

### What happened

When running make i get:
```
curl -O https://artifactory.hashdata.xyz/artifactory/greenplum/cbdb-build-dependencies/nar-maven-plugin.tar.gz
curl: (7) Failed to connect to artifactory.hashdata.xyz port 443 after 75880 ms: Couldn't connect to server
```

### What you think should happen instead

curl should succeed

### How to reproduce

Run ``make``

### Operating System

Ubuntu focal

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://cloudberrydb.org/community/coc).

**Comment by tuhaihe** [comment]
Hi @edespino could you help look at this issue? Thanks.

**Comment by zlokomatic** [comment]
Any news?

Regards,

David

**Comment by tuhaihe** [comment]
Hi @zlokomatic you need to make some tiny changes to the Makefile, like this:


```git
diff --git a/Makefile b/Makefile
index d48bbcb..271eefd 100755
--- a/Makefile
+++ b/Makefile
@@ -41,8 +41,9 @@ REGRESS_DIR = $(top_builddir)
 .PHONY: build installdirs install uninstall test localconfig targetconfig installcheck targetcheck release

 install-nar-snapshot:
-       curl -O https://artifactory.hashdata.xyz/artifactory/greenplum/cbdb-build-dependencies/nar-maven-plugin.tar.gz
+       curl -o nar-maven-plugin.tar.gz https://codeload.github.com/maven-nar/nar-maven-plugin/tar.gz/refs/tags/nar-maven-plugin-3.5.2
        tar xzf nar-maven-plugin.tar.gz
+       mv nar-maven-plugin-nar-maven-plugin-3.5.2 nar-maven-plugin
        (cd nar-maven-plugin ; mvn)
```

I will submit one PR to fix this. You can update the file manually to try to see if it works.

**Comment by tuhaihe** [comment]
PR is https://github.com/cloudberrydb/pljava/pull/2 . For the `nar-maven-plugin` build, tested successfully.

**Comment by zlokomatic** [comment]
Hi,

i tried again with updated source code.

Now the error is:

```
[INFO] Installing /usr/src/mymaven/nar-maven-plugin/target/nar-maven-plugin-3.5.2.jar to /root/.m2/repository/com/github/maven-nar/nar-maven-plugin/3.5.2/nar-maven-plugin-3.5.2.jar
[INFO] Installing /usr/src/mymaven/nar-maven-plugin/pom.xml to /root/.m2/repository/com/github/maven-nar/nar-maven-plugin/3.5.2/nar-maven-plugin-3.5.2.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.172 s
[INFO] Finished at: 2024-09-06T06:36:44Z
[INFO] ------------------------------------------------------------------------
mvn clean install
[INFO] Scanning for projects...
[WARNING] The POM for com.github.maven-nar:nar-maven-plugin:jar:3.10.2-SNAPSHOT is missing, no dependency information available
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Unresolveable build extension: Plugin com.github.maven-nar:nar-maven-plugin:3.10.2-SNAPSHOT or one of its dependencies could not be resolved:
	Could not find artifact com.github.maven-nar:nar-maven-plugin:jar:3.10.2-SNAPSHOT
 @
[ERROR] Unknown packaging: nar @ line 14, column 13
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project org.postgresql:pljava-so:1.5.0 (/usr/src/mymaven/pljava-so/pom.xml) has 2 errors
[ERROR]     Unresolveable build extension: Plugin com.github.maven-nar:nar-maven-plugin:3.10.2-SNAPSHOT or one of its dependencies could not be resolved:
[ERROR]     	Could not find artifact com.github.maven-nar:nar-maven-plugin:jar:3.10.2-SNAPSHOT
[ERROR]     -> [Help 2]
[ERROR]     Unknown packaging: nar @ line 14, column 13
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
[ERROR] [Help 2] http://cwiki.apache.org/confluence/display/MAVEN/PluginManagerException
make: *** [Makefile:50: build] Error 1
```

Kind Regards,

David

**Comment by tuhaihe** [comment]
Hey @gfphoenix78 could you help take a look at the issue update? Now it’s beyond my knowledge. Thanks.

**Comment by zlokomatic** [comment]
Thanks, @tuhaihe

By looking at the build steps, do i have to compile on the database cluster? 
As it will run gp_config and copy libs from the database!?

Kind Regards,

David

**Comment by tuhaihe** [comment]
Hi @zlokomatic, one new pr to fix this issue: #3 . Still waiting for testing, once merged, will let you know.
